### DISCUSSION TITLE
Quadruple the sample rate of a Scoppy oscilloscope

### DISCUSSION THREAD

**Discussion by fhdm-dev** [fhdm-dev/scoppy#139] [Announcements]
The latest version of the Scoppy app (v1.030) allows you to configure the latest firmware (v18) so that the maximim sample rate is now 2MS/s compared with 500kS/s previously.

Here's how to enable it:
- Go to Menu > Settings > RP2040 Settings > Max Sample Rate
- Select 2.0MS/s and then tap OK

We've released an article on [hackster.io](https://www.hackster.io/sandy-roberts2/quadrupling-the-sample-rate-of-an-rp2040-oscilloscope-5f9a74) that gives more details.
And there is the obligatory [boring documentation here](https://oscilloscope.fhdm.xyz/app-help/rp2040-max-sample-rate-setting).

**Comment by tyeth** [comment]
Thank you! ❤️

**Comment by Jason-BCIT** [comment]
It took me a while to get around to trying it out... but this is awesome! Very much enjoyed the write up on how you did it. Now I'm re-working the front end that I made for my students (that needed some upgrading anyway) to make the most of it. I'm hoping that this version will be worth sharing!

Thank you!!

**Reply by fhdm-dev** [reply]
Glad you like it :)
Yeah, I'll have to rework my front-ends too. The MCP6002 or LM358 that I normally use won't cut it. Thinking of using a TLV9062 instead - 10MHz but still dirt cheap.

**Reply by Jason-BCIT** [reply]
Thank you for mentioning that you were looking at the TLV9062... I'm not as up to speed on opamp selection as I'd like to be. I am currently using an MCP6024, which seems to be working okay but is $3.17CAD on Digikey in quantities of 100. It's the chip that is used on the DP Scope SE, which I used to make with the students so I've got some in stock, but figured if I was designing a new front end that I should use a cheaper chip. I was keeping the cost down on the old design by only using one MCP6024 and limiting the voltage ranges available on channel 2 in order to keep costs down, but I want to get a few more opamps on the board.

I wanted to keep DIP chips as an option so I ended up finding the TL974 and said, "Hey... 12MHz, rail to rail, DIP package available -- it even has an SOIC package in case I want to get my students doing some surface mount soldering."  (We can go finer, of course, but the error rate goes up. SOIC is pretty reliable for SMT newbies.)

But it turns out I hadn't checked the input range! Sure it will work at 3.3V... but that leaves a VERY narrow input voltage range!

Turns out the TLV9064 is $0.92CAD on Digikey, and just happens to match the SOIC footprint for the TL974... I don't even have to tweak my design to make it work.

So thank  you for mentioning that! It definitely saved me a big hassle!

Oh... yes... one of the real drivers behind the new design is that I didn't leave room for a battery connection on the old design. With the Pico that didn't matter... but with the Pico W, I'll set it up to run off a couple AA batteries and put a power switch on the board, too!

**Reply by Jason-BCIT** [reply]
Just wanted to mention that I got my first prototype with the TLV9064 up and running this evening. I've still got a bit of work to get the capacitors in the voltage divider tuned just right (currently have a 1-3pf variable as C10 and a 20pf fixed value as C2, but hoping to get rid of the variable cap if I can. It looks good with a 100kHz square wave. Will share more as I figure it out... schematic is here: https://a360.co/45iVoeH 

Thanks again for the suggestions... and the great software!

**Reply by Jason-BCIT** [reply]
I was feeding my signal through a CD4052B switch to select the voltage ranges and while it was working great with square waves I got a really weird "blip" of distortion in the middle of sine waves. Turns out that I should have been using the newer, low voltage version of the switch, the 74LV4052D. The signal is looking much better tonight. Happily reading 200kHz signals (sine, square and triangle) using the 2MS/s function. Will follow up with more tests, and have to make some decisions about what features I want on my front end board, but it is amazing the difference that using the right chips makes!  :-)

**Reply by BlancoMad** [reply]
> Just wanted to mention that I got my first prototype with the TLV9064 up and running this evening. I've still got a bit of work to get the capacitors in the voltage divider tuned just right (currently have a 1-3pf variable as C10 and a 20pf fixed value as C2, but hoping to get rid of the variable cap if I can. It looks good with a 100kHz square wave. Will share more as I figure it out... schematic is here: https://a360.co/45iVoeH
> 
> Thanks again for the suggestions... and the great software!

can you upload schematic to another host ?

**Reply by Jason-BCIT** [reply]
You'll find the schematic (and some more info) on my PCBWAY project page at https://www.pcbway.com/project/shareproject/Scoppy_Front_End_Mostly_Through_Hole_0575305f.html

**Comment by robertrichardsnp-a11y** [comment]
> The latest version of the Scoppy app (v1.030) allows you to configure the latest firmware (v18) so that the maximim sample rate is now 2MS/s compared with 500kS/s previously.
> 
> Here's how to enable it:
> 
> * Go to Menu > Settings > RP2040 Settings > Max Sample Rate
> * Select 2.0MS/s and then tap OK
> 
> We've released an article on [hackster.io](https://www.hackster.io/sandy-roberts2/quadrupling-the-sample-rate-of-an-rp2040-oscilloscope-5f9a74) that gives more details. And there is the obligatory [boring documentation here](https://oscilloscope.fhdm.xyz/app-help/rp2040-max-sample-rate-setting).

3:461 514) < support@superboxtv.shop Superbox Gravitational waves are ripples in spacetime caused by accelerating massive objects, like black holes or neutron stars. These waves are incredibly tiny, but scientists  Opt  out it l 目 Tab1 10 tabs
